hotshot-query-service/src/data_source/fetching/block.rs

```diff
@@ -96,6 +96,11 @@ where
             AvailabilityStorage<Types> + NodeStorage<Types> + PrunedHeightStorage,
         P: AvailabilityProvider<Types>,
     {
+        // Do not fetch if we are in light weight mode
+        if fetcher.leaf_only {
```

Comment:
Yep, this seems like the appropriate place for this check to me

Reply:
And also in `Storable::store` below

Reply:
ah forgot about that since I had diffferent implementation for that leaf only ds